src/navigable-group/internal.tsx

```diff
@@ -0,0 +1,168 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { forwardRef, useEffect, useImperativeHandle, useRef } from 'react';
+import clsx from 'clsx';
+
+import {
+  SingleTabStopNavigationAPI,
+  SingleTabStopNavigationProvider,
+  useMergeRefs,
+} from '@cloudscape-design/component-toolkit/internal';
+
+import { getBaseProps } from '../internal/base-component';
+import { getAllFocusables } from '../internal/components/focus-lock/utils';
+import { hasModifierKeys } from '../internal/events';
+import { KeyCode } from '../internal/keycode';
+import { circleIndex } from '../internal/utils/circle-index';
+import handleKey from '../internal/utils/handle-key';
+import { InternalNavigableGroupProps, NavigableGroupProps } from './interfaces';
+
+import styles from './styles.css.js';
+import testUtilStyles from './test-classes/styles.css.js';
+
+const InternalNavigableGroup = forwardRef(
+  (
+    {
+      children,
+      getItemKey,
+      navigationDirection = 'horizontal',
+      __internalRootRef,
+      ...props
+    }: InternalNavigableGroupProps,
+    ref: React.Ref<NavigableGroupProps.Ref>
+  ) => {
+    const baseProps = getBaseProps(props);
+    const navigationAPI = useRef<SingleTabStopNavigationAPI>(null);
+    const containerObjectRef = useRef<HTMLDivElement>(null);
+    const containerRef = useMergeRefs(containerObjectRef, __internalRootRef);
+    const focusedIdRef = useRef<string>();
+
+    useImperativeHandle(ref, () => ({
+      focus: () => {
+        const target = getNextFocusTarget();
+        if (target) {
+          target.focus();
+        }
+      },
+    }));
+
+    function getNextFocusTarget(): null | HTMLElement {
+      if (containerObjectRef.current) {
+        const focusables = getFocusablesFrom(containerObjectRef.current);
+        return focusables.find(button => getItemKey(button) === focusedIdRef.current) ?? focusables[0] ?? null;
+      }
+      return null;
+    }
+
+    function onUnregisterActive(focusableElement: HTMLElement) {
```

Comment:
Missing unit test: I don't see a unit test for the use-case when an item gets removed from the component.

Reply:
https://github.com/cloudscape-design/components/pull/3785/files#diff-547bd0732a7f5dde74c85be4380ee3f4df6b5f26367733d70cffc134d5f4c3a6R106